IdMask[] list_id_mask